mhs/common/mhs_common/state/work_description.py

```diff
@@ -45,6 +45,24 @@ class EmptyWorkDescriptionError(RuntimeError):
     pass
 
 
+async def update_status_with_retries(wdo: WorkDescription,
+                                     update_status_method,
+                                     status: MessageStatus,
+                                     retries: int):
+    attempts = 0
+    while attempts < retries + 1:
+        try:
+            await wdo.update()
+            await update_status_method(status)
+            break
+        except OutOfDateVersionError as e:
+            logger.error('0021', 'Failed attempt to update state store')
```

Comment:
Perhaps a warning level log here, indicating that we'll be retrying (and we're on x of y retries), and then an error log in the `if attempts == retries` block?